data/core/common.lua

```diff
@@ -47,22 +47,26 @@ end
 
 
 function common.color(str)
-  local r, g, b, a = str:match("#(%x%x)(%x%x)(%x%x)")
+  local r, g, b, a = str:match("^#(%x%x)(%x%x)(%x%x)(%x?%x?)$")
   if r then
     r = tonumber(r, 16)
     g = tonumber(g, 16)
     b = tonumber(b, 16)
-    a = 1
+    if a ~= "" then
+      a = tonumber(a, 16)
+    else
+      a = 0xff
+    end
```

Comment:
This could be
```lua
a = tonumber(a, 16) or 0xff
```

Reply:
Nice!
Programming in several languages at the same time leads to forgetting such beautiful solutions(